server/queues/processors/EmailsProcessor.ts

```diff
@@ -0,0 +1,100 @@
+import { NotificationEventType } from "@shared/types";
+import CollectionCreatedEmail from "@server/emails/templates/CollectionCreatedEmail";
+import CommentCreatedEmail from "@server/emails/templates/CommentCreatedEmail";
+import CommentMentionedEmail from "@server/emails/templates/CommentMentionedEmail";
+import DocumentMentionedEmail from "@server/emails/templates/DocumentMentionedEmail";
+import DocumentPublishedOrUpdatedEmail from "@server/emails/templates/DocumentPublishedOrUpdatedEmail";
+import { Notification } from "@server/models";
+import { Event, NotificationEvent } from "@server/types";
+import BaseProcessor from "./BaseProcessor";
+
+export default class NotificationsProcessor extends BaseProcessor {
+  static applicableEvents: Event["name"][] = ["notifications.create"];
+
+  async perform(event: NotificationEvent) {
+    const notification = await Notification.scope([
+      "withTeam",
+      "withUser",
+      "withActor",
+    ]).findByPk(event.modelId);
+    if (!notification) {
+      return;
+    }
+
+    const notificationId = notification.id;
+
+    switch (notification.event) {
+      case NotificationEventType.UpdateDocument:
+      case NotificationEventType.PublishDocument: {
+        await new DocumentPublishedOrUpdatedEmail(
+          {
+            to: notification.user.email,
```

Comment:
Going to let this sit for a day or so, but I think one last refactor is needed here so that the Email presenters just take the `notification` record and don't need it split out like this…